app/demo-geant-integration/ActionInitialization.cc

```diff
@@ -76,7 +76,7 @@ void ActionInitialization::Build() const
                       init_celeritas_});
     // Event action saves event ID for offloading and runs queued particles at
     // end of event
-    this->SetUserAction(new EventAction{transport});
+    this->SetUserAction(new EventAction{params_, transport});
```

Comment:
I think it's better to keep them separate because they're serving different purposes and because there's no direct relationship between the two (other than you need the shared params to build the transport and to interpret the exceptions). When we combined the params + state into CoreRef I ended up having to split them apart later, so I'd like to avoid that pitfall again.